src/events.rs

```diff
@@ -341,3 +342,91 @@ pub enum EventType {
         status_update_serial: StatusUpdateSerial,
     },
 }
+
+impl EventType {
+    pub fn to_json(&self, timestamp: Option<i64>) -> Value {
+        let mut tree: serde_json::Map<String, Value> = serde_json::Map::new();
+
+        tree.insert("event_type".to_string(), Value::Number(self.as_id().into()));
+
+        let (data1, data2) = match &self {
```

Comment:
This match is duplicate code with `dc_event_get_data1_int()`, `dc_event_get_data2_int()`, `dc_event_get_data2_str()`, should probably be unified

Reply:
those give int and string, this gives json values  number, string and null.
so when unifying, 0 could and should probably be converted to null ?

Reply:
tbh I don't understand what you mean, but I just pushed 13702cc4 which does it how I had imagined - does it look OK?